app/src/main/java/com/appunite/loudius/di/NetworkModule.kt

```diff
@@ -79,6 +79,31 @@ object NetworkModule {
         }
     }
 
+    @Provides
+    @Singleton
+    @BaseAPI
+    fun provideBaseHttpClient(
+        gson: Gson,
+        @BaseAPI baseUrl: String,
+        loggingInterceptor: HttpLoggingInterceptor,
+        authInterceptor: AuthInterceptor,
+        authFailureHandler: AuthFailureHandler,
```

Comment:
question: not related to this PR but why do we need to seperate classes for AuthInterceptor and AuthFailureHandler? Can't we just do everything inside the AuthInterceptor ?

Reply:
As I see both implementations, I think they're doing different things. Can you give me an example implementation of `AuthInterceptor` with `AuthFailureHandler` inside?

Reply:
forget it from I read the code abit more and it looks like AuthFailureHandler has authFailureFlow, which needs to be Injectable as standalone, so I guess there is no way around it.